x/stakeibc/keeper/msg_server_redeem_stake.go

```diff
@@ -59,7 +59,7 @@ func (k Keeper) RedeemStake(goCtx context.Context, msg *types.MsgRedeemStake) (*
 		return nil, sdkerrors.Wrapf(types.ErrEpochNotFound, "epoch tracker found: %s", "day")
 	}
 	senderAddr := sender.String()
-	redemptionId := fmt.Sprintf("%s.%d.%s", hostZone.ChainId, epochTracker.EpochNumber, senderAddr) // {chain_id}.{epoch}.{sender}
+	redemptionId := recordstypes.UserRedemptionRecordKeyFormatter(hostZone.ChainId, epochTracker.EpochNumber, senderAddr)
```

Comment:
Does this new redemptionId format render the checks on line 73-76 obsolete? What issues might that introduce, if a user could redeem multiple times per epoch?

Reply:
It's just a formatter, it doesn't change any of the logic (unless I'm missing something)

Reply:
Nevermind, I now see how this works 👍